### PR TITLE
feat: Removed theme buttons and replaced them with radix switch

### LINKED ISSUE #1402: Feature: Replace dark mode buttons with a radix switch.
### Type of feature

🍕 Feature

### Current behavior

Screen Shot 2022-04-01 at 6 28 00 AM

### Suggested solution

This will require [Radix to be installed](https://www.radix-ui.com/docs/primitives/overview/getting-started) in the project as a dependency. 

Examples code - https://github.com/ecklf/tailwindcss-radix/blob/main/demo/components/Switch.tsx

Example from the docs (emoji is not necessary.) 
Screen Shot 2022-04-01 at 6 36 34 AM
Screen Shot 2022-04-01 at 2 22 41 PM

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by chadstewart**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR [adds] the [feature] of changing the theme by using a switch element instead of separate buttons. 

## Related Tickets & Documents

Closes #1402

## Mobile & Desktop Screenshots/Recordings

Screenshot from 2022-04-06 20-54-58
Screenshot from 2022-04-06 20-55-03

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by 0-vortex** [comment]
Great job, one minor bug I can see is that the initial theme is still set from the code to `light/dark/system` and in the case of `system` it will load the theme switch as light when the theme is dark - switching it to dark will not have any effects

**Comment by chadstewart** [comment]
> Great job, one minor bug I can see is that the initial theme is still set from the code to `light/dark/system` and in the case of `system` it will load the theme switch as light when the theme is dark - switching it to dark will not have any effects

Ahh, I see. I'll check that out. I have a question about the system theme actually. Are we still going to support the system theme if there is an element that only has two states as opposed to the 3 states we have?

**Comment by 0-vortex** [comment]
> I have a question about the system theme actually. Are we still going to support the system theme if there is an element that only has two states as opposed to the 3 states we have?

Not right now, supporting multiple themes requires a select drop down or multi value switch acting like a radio box. They make sense when we have color a11y but not now, the way the system theme is implemented it can serve a sane default to our light/dark switch.

TL;DR: keep light/dark themes, repurpose system to provide default state to the switch

**Comment by bdougie** [comment]
> TL;DR: keep light/dark themes, repurpose system to provide default state to the switch

I was going to say this as well. Set the default with system if provided, otherwise safe to keep it a binary switch.

**Comment by bdougie** [comment]
@chadstewart can you also add this change to the storybook as well?

**Comment by chadstewart** [comment]
Completely forgot about that. Sorry about that. I'll add it to Storybook when I go to resolve the bug in this PR.

Thanks for the feedback @bdougie and @0-vortex!

**Comment by bdougie** [comment]
Going to merge this now, but opening an issue re: storybook.
